src/matrix.rs

```diff
@@ -11,10 +11,20 @@ use approxeq::ApproxEq;
 use point::Point2D;
 use num::{One, Zero};
 
-use num_lib::NumCast;
+use num_lib::{Float, NumCast};
 use std::ops::{Add, Mul, Div, Neg, Sub};
 
-pub fn Matrix4<T:Add<T, Output=T> + Clone + ApproxEq<T> + Mul<T, Output=T> + One + Zero>(
+pub fn Matrix4<T: Add<T, Output=T> +
+                  Clone +
+                  ApproxEq<T> +
+                  Mul<T, Output=T> +
+                  One +
+                  Zero +
+                  Neg<Output=T> +
+                  Div<T, Output=T> +
+                  Sub<T, Output=T> +
+                  NumCast +
+                  Float>(
```

Comment:
I wonder if we should make a trait that encompasses all of these to avoid repeating ourselves so much…